tutorials/future/torch/mnist_tutorial.py

```diff
@@ -29,149 +34,81 @@ def forward(self, x):
     x = self.fc3(x)
     return x
 
-######
-# load MNIST
-######
-train_transforms = torchvision.transforms.Compose(
-        [
-            torchvision.transforms.ToTensor(),
-        ],
-        )
-test_transforms = torchvision.transforms.Compose(
-        [
-            torchvision.transforms.ToTensor(),
-        ],
-        )
-train_dataset = torchvision.datasets.MNIST(
-    root='./data',
-    train=True,
-    transform=train_transforms,
-    download=True
-    )
-test_dataset = torchvision.datasets.MNIST(
-    root='./data',
-    train=False,
-    transform=test_transforms,
-    download=True
-    )
-train_loader = torch.utils.data.DataLoader(
-    train_dataset,
-    batch_size=128,
-    shuffle=True,
-    num_workers=2
-    )
-test_loader = torch.utils.data.DataLoader(
-    test_dataset,
-    batch_size=128,
-    shuffle=False,
-    num_workers=2
-    )
-
-######
-# build and train network
-######
-device = 'cuda' if torch.cuda.is_available() else 'cpu'
-print(device)
-
-net = LeNet5(1, 2)
-loss_fn = torch.nn.CrossEntropyLoss(reduction='mean')
-optimizer = torch.optim.Adam(net.parameters(), lr=1e-4, weight_decay=0)
-
-def train(n_epoch):
-  print('training starts')
+
+def ld_mnist():
+  train_transforms = torchvision.transforms.Compose([torchvision.transforms.ToTensor()])
+  test_transforms = torchvision.transforms.Compose([torchvision.transforms.ToTensor()])
+  train_dataset = torchvision.datasets.MNIST(root='./data', train=True, transform=train_transforms, download=True)
+  test_dataset = torchvision.datasets.MNIST(root='./data', train=False, transform=test_transforms, download=True)
+  train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=128, shuffle=True, num_workers=2)
+  test_loader = torch.utils.data.DataLoader(test_dataset, batch_size=128, shuffle=False, num_workers=2)
+  return EasyDict(train=train_loader, test=test_loader)
+
+
+def main(_):
+  device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+  # Load training and test data
+  data = ld_mnist()
+
+  # Instantiate model, loss, and optimizer for training
+  net = LeNet5(1, 2)
+  loss_fn = torch.nn.CrossEntropyLoss(reduction='mean')
+  optimizer = torch.optim.Adam(net.parameters(), lr=1e-4, weight_decay=0)
+
+  # This will be called later to report accuracy on clean and adversarial examples
+  def eval():
+    net.eval()
+    total = 0
+    total_correct = 0
+    total_correct_fgm = 0
+    total_correct_pgd = 0
+    for x, y in data.test:
```

Comment:
If using GPU, should perhaps do

x, y = x.to(device), y.to(device)

Otherwise, PyTorch throws an error if the model is already on GPU.